arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -805,3 +752,55 @@ void RocksDBMetadata::loadInitialNumberDocuments() {
   }
   return Result();
 }
+  
+RocksDBBlockerGuard::RocksDBBlockerGuard(LogicalCollection* collection)
+    : _collection(collection),
+      _trxId(TransactionId::none()) {
+  TRI_ASSERT(_collection != nullptr);
+}
+
+RocksDBBlockerGuard::~RocksDBBlockerGuard() {
+  releaseBlocker();
+}
+
+RocksDBBlockerGuard::RocksDBBlockerGuard(RocksDBBlockerGuard&& other) noexcept
+    : _collection(other._collection),
+      _trxId(other._trxId) {
+  other._trxId = TransactionId::none();
+}
+
+RocksDBBlockerGuard& RocksDBBlockerGuard::operator=(RocksDBBlockerGuard&& other) noexcept {
+  releaseBlocker();
+
+  _collection = other._collection;
+  _trxId = other._trxId;
+  other._trxId = TransactionId::none();
+  return *this;
+}
+
+rocksdb::SequenceNumber RocksDBBlockerGuard::placeBlocker(TransactionId trxId) {
+  TRI_ASSERT(!_trxId.isSet());
+
+  auto& engine = _collection->vocbase().server().getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+  rocksdb::SequenceNumber blockerSeq = engine.db()->GetLatestSequenceNumber();
+
+  if (!trxId.isSet()) {
+    trxId = TransactionId(transaction::Context::makeTransactionId());
+  }
```

Comment:
I would prefer to have a second overload without any parameters that generates the transactionId and then calls this overload.

Reply:
Done